### PR TITLE
Validate RSS links before saving them

### LINKED ISSUE #920: Validate rss link when saving

### LINKED ISSUE #1128: Validate RSS urls before saving
Validate that it is a valid RSS feed and not just some random HTML page. Perhaps add an explanation to the UI to what an rss feed is.

WIP: 

```python

def _validate_rss_urls(urls):
    # type: (set[str]) -> tuple[list[str], list[str]]
    rpcs = []
    invalid_urls = []
    valid_urls = []
    for url in urls:
        rpc = urlfetch.create_rpc(deadline=30)
        try:
            urlfetch.make_fetch_call(rpc, url)
            rpcs.append(rpc)
        except Exception as e:
            logging.debug('Error while creating fetch call for %s: %s', url, e.message)
            rpcs.append(None)
    for url, rpc in zip(urls, rpcs):
        if not rpc:
            invalid_urls.append(url)
            continue
        try:
            response = rpc.get_result()  # type: urlfetch._URLFetchResult
        except Exception as e:
            logging.debug('Error while fetching %s: %s', url, e.message)
            invalid_urls.append(url)
            continue
        if response.status_code != 200:
            invalid_urls.append(url)
        else:
            # try and load as rss and check if title, link and description exist and is set
            doc = minidom.parseString(response.content)
            try:
                for item in doc.getElementsByTagName('item'):
                    title = item.getElementsByTagName("title")[0].firstChild.nodeValue
                    url = item.getElementsByTagName("link")[0].firstChild.nodeValue
                    description = item.getElementsByTagName("description")[0].firstChild.nodeValue
                    if not title or not url or not description:
                        raise Exception('Missing title, url or description for %s' % url)
                valid_urls.append(url)
            except Exception as e:
                logging.debug('Error while validating url: %s' % e.message)
                invalid_urls.append(url)
    return valid_urls, invalid_urls


def save_rss_urls(service_user, service_identity, data):
    # type: (users.User, unicode, SolutionRssSettingsTO) -> SolutionRssScraperSettings
    valid_urls, invalid_urls = _validate_rss_urls({url for url in data.rss_urls})
    if invalid_urls:
        raise HttpBadRequestException('invalid_urls', {'urls': invalid_urls})
    return _save_rss_urls(service_user, service_identity, valid_urls, data.notify)


@ndb.transactional()
def _save_rss_urls(service_user, service_identity, urls, notify):
    # type: (users.User, unicode, list[unicode], bool) -> SolutionRssScraperSettings
    rss_settings_key = SolutionRssScraperSettings.create_key(service_user, service_identity)
    rss_settings = rss_settings_key.get()
    current_dict = {}
    if not rss_settings:
        rss_settings = SolutionRssScraperSettings(key=rss_settings_key)
    else:
        for rss_links in rss_settings.rss_links:
            if not current_dict.get(rss_links.url, False):
                current_dict[rss_links.url] = rss_links.dry_runned

    rss_settings.notify = notify
    rss_settings.rss_links = [SolutionRssLink(url=url, dry_runned=current_dict.get(url, False))
                              for url in urls]
    rss_settings.put()
    return rss_settings

```

### PR DESCRIPTION
**Pull request by lucasvanhalst**
Closes #1128
Closes #920

### CODE REVIEW

File: src/static/js/solutions/sln-settings.js [hunk 1]
```diff
@@ -1531,15 +1531,20 @@ $(function () {
...
-            success : function(data) {
-                LocalCache.broadcastRssSettings = data;
-                renderRssSettings(data);
-            },
-            error : sln.showAjaxError
+        Requests.saveRssSettings(settings, {showError: false}).then(function (data) {
+            LocalCache.broadcastRssSettings = data;
+            renderRssSettings(data);
+        }).catch(function (error) {
+            console.log(error);
```
Comment (abom) [inline src/static/js/solutions/sln-settings.js]: Debug logging?

File: src/solutions/common/bizz/settings.py [hunk 2]
```diff
@@ -198,3 +201,72 @@ def trans():
...
+            except Exception as e:
+                logging.debug('Error while validating url: %s' % e.message, exc_info=True)
+                invalid_urls.append(rss_url)
+    return valid_urls, invalid_urls
+
+
+def save_rss_urls(service_user, service_identity, data):
+    # type: (users.User, unicode, SolutionRssSettingsTO) -> SolutionRssScraperSettings
+    valid_urls, invalid_urls = _validate_rss_urls({url for url in data.rss_urls})
+    if invalid_urls:
```
Comment (abom) [inline src/solutions/common/bizz/settings.py]: IMHO, This is very confusing to the user if there's an older invalid URL(s), whenever he tries to add or remove a valid URL, an error with another unrelated (invalid) URL(s) appears.

I suggest another place for warnings in such cases, so, raise an error for new input, and add another place under the list or something to tell the user they have an invalid RSS URLs that need to be updated.

**Comment by abom** [comment]
Cannot delete old urls, can you check this @lucasvanhalst?

**Comment by lucasvanhalst** [comment]
Right I forgot to test that

**Pull request merged**